gef.py

```diff
@@ -741,6 +743,62 @@ def __getitem__(self, item):
         return getattr(self, item)
 
 
+class GlibcHeapInfo:
+    """Glibc heap_info struct
+    See https://github.com/bminor/glibc/blob/glibc-2.34/malloc/arena.c#L64"""
+
+    def __init__(self, addr):
+        self.__addr = addr if type(addr) is int else parse_address(addr)
+        self.size_t = cached_lookup_type("size_t")
+        if not self.size_t:
+            ptr_type = "unsigned long" if current_arch.ptrsize == 8 else "unsigned int"
+            self.size_t = cached_lookup_type(ptr_type)
+
+    @property
+    def addr(self):
+        return self.__addr
+
+    @property
+    def ar_ptr_addr(self):
+        return self.addr
+
+    @property
+    def prev_addr(self):
+        return self.ar_ptr_addr + current_arch.ptrsize
+
+    @property
+    def size_addr(self):
+        return self.prev_addr + current_arch.ptrsize
+
+    @property
+    def mprotect_size_addr(self):
+        return self.size_addr + self.size_t.sizeof
+
+    @property
+    def ar_ptr(self):
+        return self.get_size_t_pointer(self.ar_ptr_addr)
+
+    @property
+    def prev(self):
+        return self.get_size_t_pointer(self.prev_addr)
+
+    @property
+    def size(self):
+        return self.get_size_t(self.size_addr)
+
+    @property
+    def mprotect_size(self):
+        return self.get_size_t(self.mprotect_size_addr)
+
+    # helper methods
+    def get_size_t_pointer(self, addr):
```

Comment:
If the function is not meant to be publicly exposed, it's better to prefix it by `__`
```suggestion
    def __get_size_t_pointer(self, addr):
```

Reply:
single underscore, not 2. doubles mangles